linter_driver_rustc/src/ast/rustc.rs

```diff
@@ -26,7 +24,61 @@ impl<'ast, 'tcx> std::fmt::Debug for RustcContext<'ast, 'tcx> {
     }
 }
 
-impl<'ast, 'tcx> DriverContext<'ast> for RustcContext<'ast, 'tcx> {}
+fn to_leaked_rustc_lint(lint: &Lint) -> &'static RustcLint {
+    Box::leak(Box::new(RustcLint {
+        name: lint.name,
```

Comment:
Another tiny nit: The name should ideally have some kind of prefix. Later, I would like something that indicates the lint crate and maybe that it comes from our tool. For now I think that `linter` would be a good prefix. Then the emission would say `` `#[warn(linter::test_lint)]` on by default``

Reply:
I think the way clippy does this is like
`name: concat!("linter::", stringify!($NAME)),`
for the line in the macro right?

Reply:
Yes :+1:, rustc also does that in the tool lint macro :upside_down_face: